### PR TITLE
10 create direct gradient access for pytorch

### PR DESCRIPTION
**Pull request by EricKolibacz**
This Pull request will fix the explanation generation based on gradient access through PyTorch.

### CODE REVIEW

**Review by tuananhroman — APPROVED** [review]

**Review by tuananhroman — APPROVED** [review]

**Review by tuananhroman — APPROVED** [review]

**Review by tuananhroman — APPROVED** [review]

**Review by tuananhroman — CHANGES_REQUESTED** [review]
mnst_tf should be kept as it is part of the tutorials

**Review by tuananhroman — APPROVED** [review]

File: src/maxi/utils/loss_utils.py [hunk 1]
```diff
@@ -184,19 +184,13 @@ def torch_extract_target_proba(P: torch.Tensor, t: int) -> torch.Tensor:
...
 
-    target_vector = torch.zeros(len(P), dtype=torch.float32)
-    for i in range(len(P)):
-        target_vector[i] = torch_extract_target_proba(P[i], t)
-    return target_vector
+    return P[:, t]
 
 
-def torch_extract_nontarget_proba(P: torch.Tensor, t: int) -> torch.Tensor:
+def torch_extract_nontarget_proba(P: torch.Tensor, t: torch.int64) -> torch.Tensor:
```
Comment (tuananhroman) [inline src/maxi/utils/loss_utils.py:192]: Does this also work when t is 0 or -1?

**Review by tuananhroman — APPROVED** [review]

**Review by tuananhroman — APPROVED** [review]

**Review by tuananhroman — APPROVED** [review]

File: src/maxi/utils/loss_utils.py (hunk 1 above)
Comment (EricKolibacz) [reply src/maxi/utils/loss_utils.py:192]: For 0 it does, for -1 most probably not. Valid values should be: [0, t.shape[1]-1]. If we keep it this way, we might have two options:

- add a safety check to be sure we are not out of bound for the value of t
- add tests to be sure the behavior is as expected (generally advisable)

**Review by tuananhroman — APPROVED** [review]

File: src/maxi/utils/loss_utils.py (hunk 1 above)
Comment (EricKolibacz) [reply src/maxi/utils/loss_utils.py:192]: Solved with commit 62c4f17

**Review by tuananhroman — APPROVED** [review]
